### PR TITLE
Update git-sync and opentelemetry images for CVE fixes

### PR DESCRIPTION
**Pull request by nan-yu**

### CODE REVIEW

**Review by karlkfi — APPROVED** [review]
/lgtm

File: manifests/templates/otel-collector.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ spec:
     spec:
       containers:
       - name: otel-collector
-        image: gcr.io/config-management-release/otelcontribcol:v0.89.0-gke.1
+        image: gcr.io/config-management-release/otelcontribcol:v0.91.0-gke.0
```
Comment (sdowell) [inline manifests/templates/otel-collector.yaml:96]: Looks like this may need a bump for `golang.org/x/crypto`

Comment (nan-yu) [reply manifests/templates/otel-collector.yaml:96]: /hold

Patched the version. I'll bump the image version later.

Comment (nan-yu) [reply manifests/templates/otel-collector.yaml]: updated

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**